src/overlays/actors/ovl_Dm_Zl/z_dm_zl.c

```diff
@@ -30,27 +30,246 @@ const ActorInit Dm_Zl_InitVars = {
+    { &gDmZl4HandsOverEmblemLoopAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -10.0f },
+    { &gDmZl4GivingItemStartAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_ONCE, -10.0f },
+    { &gDmZl4GivingItemLoopAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -10.0f },
+    { &gDmZl4RaisingOcarinaToPlayAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_ONCE, -10.0f },
+    { &gDmZl4PlayingOcarinaAnim, 1.0f, 0.0f, -1.0f, ANIMMODE_LOOP, -10.0f },
+};
+
+typedef enum {
+    ZELDA_ANIM_FACING_AWAY,
+    ZELDA_ANIM_TURNING_TOWARD_PLAYER,
+    ZELDA_ANIM_FACING_PLAYER,
+    ZELDA_ANIM_GIVING_OCARINA_START,
+    ZELDA_ANIM_GIVING_OCARINA,
+    ZELDA_ANIM_PLAYING_OCARINA_START,
+    ZELDA_ANIM_PLAYING_OCARINA,
+    ZELDA_ANIM_MAX,
+} DmZlAnimations;
+
+static TexturePtr sMouthTextures[4] = {
+    gZl4MouthNeutralTex,
+    gZl4MouthOpenSmilingTex,
+    gZl4MouthFrowningTex,
+    gZl4MouthOpenTex,
+};
+
+typedef enum {
+    /* 0 */ ZELDA_MOUTH_NEUTRAL,
+    /* 1 */ ZELDA_MOUTH_SMILING,
+    /* 2 */ ZELDA_MOUTH_FROWNING,
+    /* 3 */ ZELDA_MOUTH_OPEN,
+} DmZlMouthTextures;
+
+static TexturePtr sEyeTextures[] = {
+    gDmZl4EyeOpenNormalTex, gDmZl4EyeBlinkingTex,        gDmZl4EyeClosedTex,           gDmZl4EyeWideTex,
+    gDmZl4EyeHappyTex,      gDmZl4EyeOpenLookingLeftTex, gDmZl4EyeOpenLookingRightTex,
+};
+
+typedef enum {
+    /* 0 */ ZELDA_EYE_OPEN_NORMAL,
+    /* 1 */ ZELDA_EYE_BLINKING,
+    /* 2 */ ZELDA_EYE_CLOSED,
+    /* 3 */ ZELDA_EYE_WIDE,
+    /* 4 */ ZELDA_EYE_HAPPY,
+    /* 5 */ ZELDA_EYE_OPEN_LOOKING_LEFT,
+    /* 6 */ ZELDA_EYE_OPEN_LOOKING_RIGHT,
+} DmZlEyeTextures;
+
+void DmZl_ChangeAnimation(SkelAnime* skelAnime, AnimationInfo animation[], u16 index) {
+    f32 endFrame;
+
+    animation += index;
+    endFrame = (animation->frameCount < 0.0f) ? Animation_GetLastFrame(animation->animation) : animation->frameCount;
+
+    Animation_Change(skelAnime, animation->animation, animation->playSpeed, animation->startFrame, endFrame,
+                     animation->mode, animation->morphFrames);
+}
+
+void DmZl_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    DmZl* this = THIS;
+
+    this->animationIndex = 0;
+    this->unused2BA = 0;
+    this->actor.targetArrowOffset = 1000.0f;
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 24.0f);
+    // these three set to NULL should mean they are dynamically allocated
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gZl4Skeleton, NULL, NULL, NULL, 0);
+    DmZl_ChangeAnimation(&this->skelAnime, &sAnimations[this->animationIndex], ZELDA_ANIM_FACING_AWAY);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = DmZl_DoNothing;
+}
+
+void DmZl_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void DmZl_DoNothing(DmZl* this, GlobalContext* globalCtx) {
+}
+
+void DmZl_UpdateCutscene(DmZl* this, GlobalContext* globalCtx) {
+    s32 actionIndex;
+
+    if (Cutscene_CheckActorAction(globalCtx, 0x66)) { // currently in cutscene
+        actionIndex = Cutscene_GetActorActionIndex(globalCtx, 0x66);
```

Comment:
Please remove the comment, there are currently 122 uses of this function in the codebase, and it's clear from the name itself what the function does: it checks if the current cs action is the one in argument.

Reply:
How is it clear? What is "ActorAction?" It's the same thing as a "cs action" I guess?

At first glance I assumed it was checking what the actor or player was busy doing, what their "Action" was. Maybe they are busy in an animation, picking something up, which would make sense for a function in the Cutscene File to check if it can **start** a cutscene. Instead its checking if a cutscene is already started, which is different then my first assumption. 

And then we pass a mystery number `0x66` to it, that doesn't help explain what is happening either. Are there a finite list of Cutscenes accessible from global? A small list of limited player actions sounds more believable. 

Nothing about the name of this function alone is intuitive from this end, but I'll remove it.